terracumber/terraformer.py

```diff
@@ -72,7 +72,7 @@ def inject_repos(self, custom_repositories_json):
             except JSONDecodeError:
                 return 1
             for node in repos.keys():
-                if node == 'server' or node == 'proxy':
```

Comment:
We want to have the ability to inject custom repositories in any kind of node.
I'm not sure when we decided to limit this only to the server, but right now, we must have it on the proxy too, at least for 5.0 due to https://github.com/SUSE/spacewalk/issues/25575#issuecomment-2425937972 
https://github.com/SUSE/spacewalk/issues/25575#issuecomment-2429292573